### PR TITLE
omit URL's password when stringifying URL for segment name

### PR DESCRIPTION
**Pull request by aloababa**
*Issue #421*

*Description of changes:*

Stripping basic auth password from url to avoid leak in xray traces.

### CODE REVIEW

**Comment by atshaw43** [comment]
Can you add some unit tests for this?

**Comment by aloababa** [comment]
I've added TestRoundTripWithBasicAuth unit test.

File: xray/client_test.go [hunk 1]
```diff
@@ -177,6 +178,62 @@ func TestRoundTripWithQueryParameter(t *testing.T) {
...
+
+	seg, err := td.Recv()
+	if !assert.NoError(t, err) {
+		return
+	}
+	var subseg *Segment
+	if assert.NoError(t, json.Unmarshal(seg.Subsegments[0], &subseg)) {
+		assert.Equal(t, "remote", subseg.Namespace)
+		assert.Equal(t, http.MethodGet, subseg.HTTP.Request.Method)
+		assert.Equal(t, stripURL(*u), subseg.HTTP.Request.URL)
```
Comment (atshaw43) [inline xray/client_test.go:226]: We are testing the function against the function. We are not really testing if the functionality works. Let's test the output against a source of truth.

```suggestion
		assert.Equal(t, "http://user:***@127.0.0.1:"+u.Port(), subseg.HTTP.Request.URL)
```

**Review by atshaw43 — CHANGES_REQUESTED** [review]

**Comment by aloababa** [comment]
I've updated the pull request.

Thanks for the response.

**Review by atshaw43 — APPROVED** [review]

**Comment by atshaw43** [comment]
You will need to rebase with the upstream to merge this.

**Comment by aloababa** [comment]
I rebased on master.

**Review by wangzlei — COMMENTED** [review]
LGTM

**Pull request merged**